src/tng-hooks.src.js

```diff
@@ -213,4 +224,45 @@
 			throw new Error("useEffect() only valid inside an Articulated Function or a Custom Hook.");
 		}
 	}
+
+	function useMemo(fn, ...guards) {
+		// passed in any guards?
+		if (guards.length > 0) {
+			// only passed a single guards list?
+			if (guards.length == 1 && Array.isArray(guards[0])) {
+				guards = guards[0];
+			}
+		}
+		// no guards passed
+		// NOTE: different handling than an empty guards list like []
+		else {
+			guards = undefined;
+		}
+
+		var bucket = getCurrentBucket();
+		if (bucket) {
+			// need to create this memoization-slot for this bucket?
+			if (!(bucket.nextMemoizationIdx in bucket.memoizations)) {
+				bucket.memoizations[bucket.nextMemoizationIdx] = [];
+			}
+
+			let memoizationIdx = bucket.nextMemoizationIdx;
+			let memoization = bucket.memoizations[memoizationIdx];
+
+			// check guards?
+			if (guardsChanged(memoization[1], guards)) {
+				// invoke the memoization
+				memoization[0] = fn();
```

Comment:
```suggestion
 	 	 	 	try {
				 	memoization[0] = fn();
 	 	 	 	}
 	 	 	  	finally {
```

Reply:
Note that this is the first place where some of the changes I mentioned in the main review message above would go. Line 256 here would likely look like `memoization[0] = fn;` (not `fn();`).